internal/server/analytics/clickhouse/client.go

```diff
@@ -0,0 +1,190 @@
+package clickhouse
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"go.flipt.io/flipt/internal/config"
+	fliptsql "go.flipt.io/flipt/internal/storage/sql"
+	"go.flipt.io/flipt/rpc/flipt/analytics"
+	"go.uber.org/zap"
+)
+
+// Step defines the value and interval name of the time windows
+// for the Clickhouse query.
+type Step struct {
+	intervalValue int
+	intervalStep  string
+}
+
+var dbOnce sync.Once
+
+const (
+	counterAnalyticsTable = "flipt_counter_analytics"
+	counterAnalyticsName  = "flag_evaluation_count"
+	timeFormat            = "2006-01-02 15:04:05"
+)
+
+type Client struct {
+	conn         *sql.DB
+	forceMigrate bool
+}
+
+// New constructs a new clickhouse client that conforms to the analytics.Client contract.
+func New(logger *zap.Logger, cfg *config.Config, forceMigrate bool) (*Client, error) {
+	var (
+		conn          *sql.DB
+		clickhouseErr error
+	)
+
+	dbOnce.Do(func() {
+		err := runMigrations(logger, cfg, forceMigrate)
+		if err != nil {
+			clickhouseErr = err
+			return
+		}
+
+		connection, err := connect(cfg.Analytics.Clickhouse.URL)
+		if err != nil {
+			clickhouseErr = err
+			return
+		}
+
+		conn = connection
+	})
+
+	if clickhouseErr != nil {
+		return nil, clickhouseErr
+	}
+
+	return &Client{conn: conn, forceMigrate: forceMigrate}, nil
+}
+
+// runMigrations will run migrations for clickhouse if enabled from the client.
+func runMigrations(logger *zap.Logger, cfg *config.Config, forceMigrate bool) error {
+	m, err := fliptsql.NewMigrator(*cfg, logger, true)
+	if err != nil {
+		return err
+	}
+
+	if err := m.Up(forceMigrate); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func connect(connectionString string) (*sql.DB, error) {
+	conn := clickhouse.OpenDB(&clickhouse.Options{
+		Addr: []string{connectionString},
+		Auth: clickhouse.Auth{
+			Database: "flipt_analytics",
+		},
+	})
+
+	if err := conn.Ping(); err != nil {
+		return nil, err
+	}
+
+	return conn, nil
+}
+
+func (c *Client) GetFlagEvaluationsCount(ctx context.Context, req *analytics.GetFlagEvaluationsCountRequest) ([]string, []float32, error) {
+	fromTime, err := time.Parse(timeFormat, req.From)
```

Comment:
I think this is equivalent to the constant in the `time` package. You can drop the constant from this package too then.
```suggestion
	fromTime, err := time.Parse(time.DateTime, req.From)
```